PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -315,7 +305,14 @@ function transform(sourcePath: string, sheetText: string, outPath: string, trans
             destination: "serialized"
         };
         const result = SaxonJS.transform(options);
-        fs.writeFileSync(outPath, result.principalResult);
+        if (hasRuleDocUrls) {
```

Comment:
We need to make sure all promises are resolved before we modify the generated SARIF report to include rule doc url.